middle_end/flambda/types/env/aliases.ml

```diff
@@ -16,6 +16,25 @@
 
 module Const = Reg_width_things.Const
 
+let map_inter map1 map2 =
+  Name.Map.merge (fun _elt a b ->
+    match a, b with
+    | None, None | Some _, None | None, Some _ -> None
+    | Some (), Some () -> Some ()
+  )
+    map1
+    map2
+
+let map_union map1 map2 =
+  Name.Map.union (fun _elt a b ->
+    match a, b with
+    | (), () -> Some ()
+  )
+    map1
+    map2
+
+let print_unit ppf () =  Format.pp_print_string ppf "()"
```

Comment:
These utility functions should go elsewhere (in `Map` perhaps).  There is also a `Unit` module already I think which has a printer.

Reply:
That's one of the function I've introduced while splitting the code -- it is a placeholder for something else and is being deleted in one of the following PR.

Reply:
@mshinwell In the standard library? `stdlib/unit.mli` only has `equal`, `compare`, and `to_string`.

Reply:
There is another `Unit` in the `middle_end` hierarchy.

Reply:
Ah, there it is.

Reply:
Replaced all uses of `print_unit` with `Unit.print`. Keeping the other functions because they're actually specific to the module (or they will be when they're replaced with the `Map_to_canonical` module).